### ISSUE TITLE
App authentication doesn't work with provided action

### ISSUE THREAD

**Issue by allanlewis** [mszostok/codeowners-validator#162]
**Description**

I created a Github app for use with the action provided by this repo but I get the following error:

```
while creating GitHub client: GitHub authorization is required, provide ACCESS_TOKEN or APP_ID
```

My step is something like this:

```yaml
- name: Run CODEOWNERS validator
  uses: mszostok/codeowners-validator@v0.7.4
  with:
    checks: "owners,duppatterns,syntax"
    owner_checker_owners_must_be_teams: "true"  # Disallow individual users as owners
  env:
    GITHUB_APP_ID: ${{ secrets.REDACTED }}
    GITHUB_APP_INSTALLATION_ID: ${{ secrets.REDACTED }}
    GITHUB_APP_PRIVATE_KEY: ${{ secrets.REDACTED }}
```

**Troubleshooting**

Using the Docker image directly with the same credentials works:

```yaml
- name: Run CODEOWNERS validator
  uses: docker://mszostok/codeowners-validator:v0.7.4
  env:
    GITHUB_APP_ID: ${{ secrets.REDACTED }}
    GITHUB_APP_INSTALLATION_ID: ${{ secrets.REDACTED }}
    GITHUB_APP_PRIVATE_KEY: ${{ secrets.REDACTED }}
    REPOSITORY_PATH: "."
    OWNER_CHECKER_REPOSITORY: REDACTED/REDACTED
    OWNER_CHECKER_OWNERS_MUST_BE_TEAMS: "true"
    CHECKS: duppatterns,owners,syntax
```

**Comment by allanlewis** [comment]
Thanks, @mszostok - the reason I was trying to use env vars was that the app authentication parameters weren't listed in `action.yml`. I see that you've addressed that in ccda6cc - are you intending to make a new release in the near future?

**Comment by mszostok** [comment]
Not sure about that. However you can use it in that way without the release. You will just see the notice about missing props but the action will work properly 👍 
See: https://github.com/GitHubCODEOWNERS/codeowners-samples/actions/runs/2655128900

**Comment by iamstarkov** [comment]
im having the similar issue, I configured the actions in this way:

```
    - uses: mszostok/codeowners-validator@v0.7.4
      with:
        github_app_id: 'REDACTED'
        github_app_installation_id: 'REDACTED'
        github_app_private_key: ${{ secrets.GH_APP_PRIVATE_KEY }}
        checks: "files,owners,duppatterns,syntax"
        experimental_checks: "notowned,avoid-shadowing"
```

and I got this output where action complains that documented required fields are not expected.

```
$ Run mszostok/codeowners-validator@v0.7.4
Warning: Unexpected input(s) 'github_app_id', 'github_app_installation_id', 'github_app_private_key', valid inputs are ['entryPoint', 'args', 'github_access_token', 'github_base_url', 'github_upload_url', 'experimental_checks', 'checks', 'repository_path', 'check_failure_level', 'not_owned_checker_skip_patterns', 'owner_checker_repository', 'owner_checker_ignored_owners', 'owner_checker_allow_unowned_patterns', 'owner_checker_owners_must_be_teams', 'not_owned_checker_subdirectories', 'not_owned_checker_trust_workspace']
```

am I missing something?

**Comment by iamstarkov** [comment]
I wouldn't classify it as a warning, due to it failing the build haha.

@mszostok thank you

**Comment by mszostok** [comment]
are you sure? I tested that and the build was not failing, see: https://github.com/GitHubCODEOWNERS/codeowners-samples/actions/runs/2655128900

**Comment by iamstarkov** [comment]
@mszostok nevermind, im a dumbass and forgot to add private key in secrets. thank you again for the help
